contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8693,20 +8752,32 @@ handleClrUdtFuncCall(TSqlParser::Clr_udt_func_callContext *ctx)
 		for (size_t i = 0; i < method_calls.size(); ++i)
 		{
 			TSqlParser::Method_callContext *method = method_calls[i];
+			size_t ind = -1;
+			if (i == 0)
+			{
+				if(ctx->local_id()) ind = ctx->local_id()->stop->getStopIndex();
+				else if(ctx->subquery()) ind = ctx->subquery()->stop->getStopIndex();
+				else if(ctx->function_call()) ind = ctx->function_call()->stop->getStopIndex();
+				else if(ctx->RR_BRACKET()) ind = ctx->RR_BRACKET()->getSymbol()->getStopIndex();
+			}
+			else ind = method_calls[i-1]->stop->getStopIndex();
+
 			/* rewriting the query in case of geospatial function calls */
-			if(method->spatial_methods())
+			if (method->spatial_methods())
 			{
-				size_t ind = -1;
-				if (i == 0)
-				{
-					if(ctx->local_id()) ind = ctx->local_id()->stop->getStopIndex();
-					else if(ctx->subquery()) ind = ctx->subquery()->stop->getStopIndex();
-					else if(ctx->function_call()) ind = ctx->function_call()->stop->getStopIndex();
-					else if(ctx->RR_BRACKET()) ind = ctx->RR_BRACKET()->getSymbol()->getStartIndex();
-				}
-				else ind = method_calls[i-1]->stop->getStopIndex();
 				rewrite_geospatial_query_helper(ctx, method, ind);
 			}
+
+			/* rewriting the query in case of xml function calls */
+			if (method->xml_methods())
```

Comment:
Here as well, Since the rule is defined as `expression_list?`, we have to check the presence of expression_list.

Reply:
we are already checking the presence of `expression_list?` and throwing appropriate error in function `validateXMLFunctionArgs`.